src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/NativeRuntimeEventSource.cs

```diff
@@ -51,13 +51,13 @@ internal unsafe void ProcessEvent(uint eventID, uint osThreadID, DateTime timeSt
             }
 
             // Make sure the eventID is valid.
-            if (eventID >= m_eventData!.Length)
+            if (!m_eventData!.ContainsKey((int)eventID))
```

Comment:
Instead of this, we can check that `CollectionsMarshal.GetValueRefOrNullRef` called on next line does not return null byref. Saves extra lookup.